apiClient.go

```diff
@@ -55,35 +62,28 @@ type LWAPIRes interface {
 	HadError() bool
 }
 
-// New takes a *viper.Viper, and gives you a *Client. If there's an error, it is returned.
+// New takes a *LWAPIConfig, and gives you a *Client. If there's an error, it is returned.
 // When using this package, this should be the first function you call. Below is an example
 // that demonstrates creating the *viper.Viper and passing it to New.
 //
 // Example:
-//	config := viper.New()
-//	config.SetConfigName("lwApi")
-//	config.AddConfigPath(".")
-//	// Match environment variables as well
-//	config.AutomaticEnv()
-//	if viperErr := config.ReadInConfig(); viperErr != nil {
-//		panic(viperErr)
+//	config := LWAPIConfig{
+//		Username: "ExampleUsername",
+//		Password: "ExamplePassword",
+//		Url:      "api.liquidweb.com",
 //	}
-//	apiClient, newErr := lwApi.New(config)
+//	apiClient, newErr := lwApi.New(&config)
 //	if newErr != nil {
 //		panic(newErr)
 //	}
-func New(config *viper.Viper) (*Client, error) {
-	if err := santizeConfig(config); err != nil {
+func New(config *LWAPIConfig) (*Client, error) {
+	if err := processConfig(config); err != nil {
 		return nil, err
 	}
-	timeout := config.GetInt("lwApi.timeout")
-	if timeout == 0 {
-		timeout = 20
-	}
 
-	httpClient := &http.Client{Timeout: time.Duration(time.Duration(timeout) * time.Second)}
+	httpClient := &http.Client{Timeout: time.Duration(time.Duration(config.Timeout) * time.Second)}
 
-	if config.GetBool("lwApi.secure") != true {
+	if config.Secure != true {
```

Comment:
zero value for a bool is `false`. Due to this, we might want to change `Secure` to `Insecure` in the `LWAPIConfig`, so the default is secure if it isn't passed.

Reply:
sounds good.